services/apps/alcs/src/alcs/tag/tag.controller.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpException,
+  HttpStatus,
+  Param,
+  Patch,
+  Post,
+  Query,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiOAuth2 } from '@nestjs/swagger';
+import * as config from 'config';
+import { RolesGuard } from '../../common/authorization/roles-guard.service';
+import { UserRoles } from '../../common/authorization/roles.decorator';
+import { TagService } from './tag.service';
+import { AUTH_ROLE } from '../../common/authorization/roles';
+import { TagDto } from './tag.dto';
+import { QueryFailedError } from 'typeorm';
+
+@Controller('tag')
+@ApiOAuth2(config.get<string[]>('KEYCLOAK.SCOPES'))
+@UseGuards(RolesGuard)
+export class TagController {
+  constructor(private service: TagService) {}
+
+  @Get('/:pageIndex/:itemsPerPage')
```

Comment:
In pagination and filtering, query parameters would be used instead of URL parameters. 

`/?pageIndex=1&itemsPerPage=10`